packages/db/scripts/migrations/migrate-quizzes-to-v2.test.ts

```diff
@@ -0,0 +1,274 @@
+    });
+
+    it("should handle data without lessonPlan", async () => {
+      const dataWithoutLessonPlan = {
+        someOtherField: "value",
+      };
+
+      await expect(
+        upgradeQuizzes({
+          data: dataWithoutLessonPlan,
+          persistUpgrade: null,
+        }),
+      ).rejects.toThrow("Invalid data structure for quiz upgrade");
+    });
+
+    it("should handle mixed V1 and V2 quizzes", async () => {
+      const mixedData = {
+        lessonPlan: {
+          title: "Test Lesson",
+          starterQuiz: [
+            {
+              question: "V1 Quiz",
+              answers: ["Answer"],
+              distractors: ["Wrong"],
+            },
+          ],
+          exitQuiz: {
+            version: "v2",
+            questions: [
+              {
+                questionType: "multiple-choice",
+                question: "V2 Quiz",
+                answers: ["Answer"],
+                distractors: ["Wrong"],
+              },
+            ],
+          },
+        },
+      };
+
+      const result = await upgradeQuizzes({
+        data: mixedData,
+        persistUpgrade: null,
+      });
+
+      expect(result.wasUpgraded).toBe(true);
+
+      if (!hasLessonPlan(result.data)) {
+        throw new Error("Expected result.data to have lessonPlan");
+      }
+      const lessonPlan = result.data.lessonPlan;
+      // Starter quiz should be upgraded
+      expect((lessonPlan.starterQuiz as { version: string }).version).toBe(
+        "v2",
+      );
+      // Exit quiz should remain unchanged
+      expect(lessonPlan.exitQuiz).toEqual(mixedData.lessonPlan.exitQuiz);
+    });
+
+    it("should call persistUpgrade callback when provided", async () => {
+      const mockPersist = jest.fn(() => Promise.resolve());
+      const v1Data = {
+        lessonPlan: {
+          starterQuiz: [
+            {
+              question: "Q1",
+              answers: ["A1"],
+              distractors: ["D1"],
+            },
+          ],
+        },
+      };
+
+      await upgradeQuizzes({
+        data: v1Data,
+        persistUpgrade: mockPersist as unknown as
+          | ((upgradedData: unknown) => Promise<void>)
+          | null,
+      });
+
+      expect(mockPersist).toHaveBeenCalledTimes(1);
+      expect(mockPersist).toHaveBeenCalledWith(
+        expect.objectContaining({
+          lessonPlan: expect.objectContaining({
+            starterQuiz: expect.objectContaining({
+              version: "v2",
+            }),
+          }),
+        }),
+      );
+    });
+
+    it("should handle empty quizzes", async () => {
+      const emptyQuizData = {
+        lessonPlan: {
+          title: "Test Lesson",
+          starterQuiz: [],
+          exitQuiz: [],
+        },
+      };
+
+      const result = await upgradeQuizzes({
+        data: emptyQuizData,
+        persistUpgrade: null,
+      });
+
+      expect(result.wasUpgraded).toBe(true);
+
+      if (!hasLessonPlan(result.data)) {
```

Comment:
Would be good if you could narrow types with `expect` 

Although can you run tests without exiting at first fail in the test? In which case that could cause issues

Reply:
you could use expect and then an invariant. I'll try that

Reply:
This is fine for now!